src/miner.cpp

```diff
@@ -342,6 +366,24 @@ int BlockAssembler::UpdatePackagesForAdded(const CTxMemPool::setEntries& already
     return nDescendantsUpdated;
 }
 
+void BlockAssembler::UpdatePackagesForRemoved(const CTxMemPool::setEntries& removed,
+        indexed_modified_transaction_set &mapModifiedTx)
+{
+    for (const CTxMemPool::txiter it : removed) {
+        CTxMemPool::setEntries descendants;
+        mempool.CalculateDescendants(it, descendants);
+        for(CTxMemPool::txiter desc : descendants) {
+            if (removed.count(desc)) {
```

Comment:
FYI I've reworked this code in the latest version, combining the removal of transactions and updating `mapModifiedTx`, as this was needlessly inefficient (with an extra call to `CalculateDescendants`).

But to answer your question: `mapModifiedTx` contains transactions that are not in the constructed block which have ancestors that are in the block.  When we remove a transaction from the block, we need to remove any in-block descendants (which won't be in `mapModifiedTx`), but we need to update the ancestor feerate score for the entries in `mapModifiedTx`, so that the packages are properly sorted.

Since we generally should never select those transactions on the next call to `addPackageTxs` (outside of reorg situations, the arrival time of a descendant should be later than the arrival time of the parent), it might be better to just remove those descendants from `mapModifiedTx` rather than correct the package information, but I think its simpler to reason about the code if `mapModifiedTx` is always sorted correctly.